### PR TITLE
[Issue-16487] Add Stop for Search Indexing

### LINKED ISSUE #16487: [SearchINdexing] Allow App run to be stopped while current run is going on
**Is your feature request related to a problem? Please describe.**

Currently when we start the search indexing , more triggers cannot be invoked , since a run might be in progress, we need to add Stop current Execution functionality as well, so if a run is in progress that can be stopped,.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mohityadav766**
### Describe your changes:

Fixes #16487
- Allow Jobs to be stopped 

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java [hunk 1]
```diff
@@ -261,4 +261,32 @@ public void triggerOnDemandApplication(App application) {
...
+      JobDetail jobDetailOnDemand =
+          scheduler.getJobDetail(
+              new JobKey(
+                  String.format("%s-%s", application.getName(), ON_DEMAND_JOB), APPS_JOB_GROUP));
+
+      if (jobDetailOnDemand != null) {
+        scheduler.interrupt(jobDetailOnDemand.getKey());
+      }
+    } catch (ObjectAlreadyExistsException ex) {
+      throw new UnhandledServerException("Job is already running, please wait for it to complete.");
```
Comment (sushi30) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java:287]: This is probably not the correct message for this case.

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java [hunk 2]
```diff
@@ -261,4 +261,32 @@ public void triggerOnDemandApplication(App application) {
...
+
+  public void stopApplicationRun(App application) {
+    if (application.getFullyQualifiedName() == null) {
+      throw new IllegalArgumentException("Application's fullyQualifiedName is null.");
+    }
+    try {
+      // Interrupt any scheduled job
+      JobDetail jobDetailScheduled =
+          scheduler.getJobDetail(new JobKey(application.getName(), APPS_JOB_GROUP));
+      if (jobDetailScheduled != null) {
```
Comment (sushi30) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java:273]: can we add debug logs here for each of the sops?

**Review by sushi30 — CHANGES_REQUESTED** [review]
Some inline comments. Can we add a test for this functionality?

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java (hunk 2 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java:273]: added

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java (hunk 1 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java]: updated @sushi30

**Comment by mohityadav766** [comment]
> Some inline comments. Can we add a test for this functionality?

addressed

**Review by sushi30 — DISMISSED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**